pkg/test-infra/tidb/operation.go

```diff
@@ -108,23 +111,36 @@ func (o *Ops) GetNodes() ([]cluster.Node, error) {
 func (o *Ops) GetClientNodes() ([]cluster.ClientNode, error) {
 	var clientNodes []cluster.ClientNode
 
-	ips, err := util.GetNodeIPs(o.cli, o.ns, map[string]string{"app.kubernetes.io/instance": o.name})
-	if err != nil {
-		return clientNodes, err
-	} else if len(ips) == 0 {
-		return clientNodes, errors.New("k8s node not found")
-	}
-
-	svc, err := o.getTiDBServiceByMeta(&o.tc.Service.ObjectMeta)
-	if err != nil {
-		return clientNodes, err
+	if util.IsInK8sPodEnvironment() {
+		svc, err := o.getTiDBServiceByClusterName(o.ns, o.name)
+		if err != nil {
+			return nil, err
+		}
+		clientNodes = append(clientNodes, cluster.ClientNode{
+			Namespace:   o.ns,
+			ClusterName: o.name,
+			IP:          svc.Spec.ClusterIP,
+			Port:        getTiDBServicePort(svc),
```

Comment:
Maybe we shoud raise an error If the `getTiDBServicePort` returns 0?

Reply:
Resolved on 7c88cd78099cc909c51233ae5538d27172fcac22. For convenient, panic if not expect.